core/src/main/kotlin/ch/ergon/dope/resolvable/clause/model/JoinClause.kt

```diff
@@ -21,43 +21,75 @@ private enum class JoinType(val type: String) {
 }
 
 sealed class SelectJoinClause : ISelectJoinClause {
-    private val dopeQuery: DopeQuery
+    private val joinType: JoinType
+    private val bucket: Bucket
+    private var onCondition: TypeExpression<BooleanType>? = null
+    private var onKeys: Field<out ValidType>? = null
+    private var onKey: Field<out ValidType>? = null
+    private var forBucket: Bucket? = null
```

Comment:
make all of them val and initialize it in the constructors

Reply:
done